src/wallet/components/wallet-advanced-send/components/SendAddressInput.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import { TextInput } from '@/internal/components/TextInput';
+import { background, border, cn, color } from '@/styles/theme';
+import { useCallback, useMemo } from 'react';
+import type { SendAddressInputProps } from '../types';
+import { resolveAddressInput } from '../utils/resolveAddressInput';
+import { validateAddressInput } from '../utils/validateAddressInput';
+
+export function SendAddressInput({
+  selectedRecipientAddress,
+  recipientInput,
+  setRecipientInput,
+  setValidatedInput,
+  handleRecipientInputChange,
+  classNames,
+}: SendAddressInputProps) {
+  const displayValue = useMemo(() => {
+    if (selectedRecipientAddress?.display) {
+      return selectedRecipientAddress.display;
+    }
+    return recipientInput;
+  }, [selectedRecipientAddress, recipientInput]);
```

Comment:
suggestion: Since the the values are just strings, it might be cleaner to skip the memo and just

```tsx
const displayValue = selectedRecipientAddress?.display || recipientInput;
```

Reply:
good call, thanks